geoutils/georaster.py

```diff
@@ -1250,41 +1254,67 @@ def interp_points(self,pts,nsize=1,mode='linear',band=1):
         assert mode in ['mean', 'linear', 'cubic', 'quintic',
                         'nearest'], "mode must be mean, linear, cubic, quintic or nearest."
 
-        rpts = []
+        # get coordinates
+        x, y = list(zip(*pts))
 
-        #TODO: might need to check if coordinates are center or point in the metadata here...
+        # if those are in latlon, convert to Raster crs
+        if input_latlon:
+            init_crs = pyproj.CRS(4326)
+            dest_crs = pyproj.CRS(self.crs)
+            transformer = pyproj.Transformer.from_crs(init_crs,dest_crs)
+            x, y = transformer.transform(x,y)
 
+        i, j = self.ds.index(x, y, op=np.float32)
+        # # necessary because rio.Dataset.xy does not return Iterable for a single point
+        if not isinstance(i, Iterable):
+            i, j = (np.asarray([i,]), np.asarray([j,]))
+        else:
+            i, j = (np.asarray(i), np.asarray(j))
```

Comment:
what's the difference between np.array and np.asarray?

Reply:
I think np.asarray is the official one for convert all types of Iterables: https://numpy.org/doc/stable/reference/generated/numpy.asarray.html
Don't really know the difference, maybe @erikmannerfelt can help here

Reply:
From the documentation linked by @rhugonnet:

> Array interpretation of a. No copy is performed if the input is already an ndarray with matching dtype and order. If a is a subclass of ndarray, a base class ndarray is returned.

It's a zero-copy operation, assuming the array is already an array. Otherwise it will convert to an array. Calling `np.array(arr)` makes for potential copying, which is unnecessary overhead.

Reply:
Good to know!